src/utils/auth.ts

```diff
@@ -0,0 +1,40 @@
+import express, {Request, NextFunction} from 'express';
+const jwt = require('jsonwebtoken');
+import { PrismaClient } from '@prisma/client';
+
+const prisma = new PrismaClient();
+
+
+export interface ReqWithUser extends Request {
+    user: {
+        id: number;
+        email: string;
+        dateCreated: Date;
+        username: string;
+        password?: string;
+    }
+}
+
+// TODO: Test out this middleware once login endpoint is done and returns a JSON web token
+const authenticateJWT = async (req: ReqWithUser, next: NextFunction, prisma: PrismaClient) => {
+    const authHeader = req.headers.authorization;
+
+    if (authHeader) {
+        const token = authHeader.split(' ')[1];
+        try {
+            const user = await jwt.verify(token, process.env.ACCESS_TOKEN_SECRET)
+            req.user = await prisma.user.findUniqueOrThrow({
+                where: {
+                    id: user.id
+            }
+        })
+
+        delete req.user.password;
```

Comment:
nit: just a suggestion for style, not a required change

```suggestion
            const {id} = await jwt.verify(token, process.env.ACCESS_TOKEN_SECRET)
            const user = await prisma.user.findUniqueOrThrow({
                where: {
                    id,
            }
        })

        delete user.password;
        
        req.user = user;
```